### ISSUE TITLE
[FEATURE] Allow projectiles for certain bosses to land via config

### ISSUE THREAD

**Issue by TranquilStar** [mariumbacchus/Soulslike-Weaponry#169]
**Is your feature request related to a problem? Please describe.**
this one is a yes and no, so I'm trying to make a RPG related modpack for a server, in version 1.20.1. now, I'm testing bosses tryna balance things out and make sure certain things are okay before making things official, I added this mod as a means to add more rpg elements and bosses.

**Describe the solution you'd like**

This is what I would like, a configuration for certain bosses that are immune to projectiles, respectively the fallen icon and harbinger are immune to arrow projectiles as a example, as I was testing the archer class and you basically cant be a archer (if they are wanting to main a bow only, and a side note all abilities they have are bow only), they are forced to go melee, which if they made their build right, they wont be of much help because they will be squishy (kinda, I'm doing a leveling system but that still needs balancing ofcourse).

that's for archers however, my witches, so I'm not sure if I also have to talk to the bewitchment dev team as well or if certain bosses just are flat out immune to potions (vanilla magic), they do work with the rpg spells like the frost, arcane, fire etc. but there may be a possibility that there may be a few select that would like to main or use the potions to fight bosses as well, the harm effect is null for fallen icon and their other half, but I assumed it was due to being undead atleast he looks of it, but wither and poison work, if harming doesnt work could possibly maybe make him take damage via instant health pots if not harming in a config? I brought up bewitchment as it's the witchcraft mod of the pack but I'm pretty sure it uses vanilla potion effects so I assume it's just them being immune to the effects overall.

I havent seen anything else yet but I'll probably add incase I find anything.

so in short, a config to allow potion effects to work on bosses as well as projectiles such as arrows and fire works to work against bosses so that everyone and everything has a place to shine in the class system for the modpack.

**Comment by mariumbacchus** [comment]
Alright, will add the config lines for projectile immunity, fire immunity and such. (Almost) all the bosses are immune to fire, and yes the Returning Knight is undead so healing potions should damage it, while the Fallen Icon (Harbinger) is not undead.

**Comment by TranquilStar** [comment]
thankyou so much! the fire damage thing I kinda understand and plus it's not a major issue, even with the immunity the fire wizard can still do their magic on them as it's just a flat damage (no burn after effect which is interesting)

thisll definitely help bring things together so everyone has a spot to shine

thankyou again!~

**Comment by mariumbacchus** [comment]
Added the config lines, will be live next release.
